common/test/keyboards/build.sh

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env bash
+
+set -e
+set -u
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../../resources/build/build-utils.sh"
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+THIS_DIR="$(dirname "$THIS_SCRIPT")"
+
+display_usage() {
+  echo "usage: build.sh [build options] [targets]"
+  echo
+  echo "Build options:"
+  echo "  --clean, -c       Clean instead of build"
+  echo "  --debug, -d       Debug build"
+  echo "  --silent, -s      Suppress information messages"
+  echo "  --keyboard, -k    Build only keyboards (not packages)"
+  echo "  --kmcomp path     Specify path to kmcomp.exe, defaults"
+  echo "                    to windows/bin/developer/kmcomp.exe"
+  echo
+  echo "Targets (all unless specified):"
+
+  for d in "$THIS_DIR/"*/; do
+    d="$(basename "$d")"
+    echo "  $d"
+  done
+
+  echo
+  exit 0
+}
+
+QUIET=false
+DEBUG=false
+CLEAN=false
+KEYBOARDS_ONLY=false
+KMCOMP="$KEYMAN_ROOT/windows/bin/developer/kmcomp.exe"
+TARGETS=()
+
+# Parse args
+shopt -s nocasematch
+
+while [[ $# -gt 0 ]] ; do
+  key="$1"
+  case $key in
+    --help|-h|-\?)
+      display_usage
+      ;;
+    --clean|-c)
+      CLEAN=true
+      ;;
+    --debug|-d)
+      DEBUG=true
+      ;;
+    --silent|-s)
+      QUIET=true
+      ;;
+    --keyboard|-k)
+      KEYBOARDS_ONLY=true
+      ;;
+    --kmcomp)
+      shift
+      KMCOMP="$1"
+      ;;
+    *)
+      TARGETS+=("$key")
+  esac
+  shift
+done
+
+# TODO: while this is intended to be cross platform, we
+# don't currently have a binary version of kmcomp available
+# during Linux and macOS builds, so that will need to be
+# manually sourced.
+
+case "${OSTYPE}" in
+  "cygwin")
+    KMCOMP_LAUNCHER=
+    ;;
+  "msys")
+    KMCOMP_LAUNCHER=
+    ;;
+  "darwin"*)
+    # For Catalina (10.15) onwards, must use wine64
+    base_macos_ver=10.15
+    macos_ver=$(sw_vers -productVersion)
+    if verlt "$macos_ver" "$base_macos_ver"; then
+      KMCOMP_LAUNCHER=wine
+    else
+      # On Catalina, and later versions:
+      # wine-4.12.1 works; wine-5.0, wine-5.7 do not.
+      # retrieve these from:
+      # `brew tap gcenx/wine && brew install --cask --no-quarantine wine-crossover`
+      # may also need to `sudo spctl --master-disable`
+      KMCOMP_LAUNCHER=wine64
+      KMCOMP="$(dirname $KMCOMP)/kmcomp.x64.exe"
+    fi
+    ;;
+  *)
+    KMCOMP_LAUNCHER=wine
```

Comment:
We should check for a `kmcomp` script (that hopefully launches kmcomp.exe) and only if that fails set `KMCOMP_LAUNCHER=wine` (see common/core/desktop/tests/unit/kmx/meson.build)

Reply:
Yeah... I'm not sure it gets us very far; this script is not going to be ideal on platforms other than Windows for now, because these keyboards should really be built with the version of kmcomp found in the current branch (which is difficult currently on linux, macos; this will be improved once kmcompx lands). I don't want to search for another `kmcomp` script on the path automatically.

But I've added this to #6358: if you pass `--kmcomp` on the command line, you are then responsible for also including `wine`.